### DISCUSSION TITLE
Is anyone using CKAN as an internal datasetcatalog?

### DISCUSSION THREAD

**Discussion by hakimsjo** [ckan/ckan#7419] [Q&A]
In Statistics Sweden we are thinking about using CKAN for tracking internal datasets for internal sharing and for external sharing. Is anyone using CKAN for internal purposes?

**Comment by pdelboca** [comment]
Hello!

I have been working on an CKAN implementation for internal purposes for the last couple of years. I suits very well our requirements.

Even when CKAN has been designed for a public interface we didn't find any issue with the opposite.

We basically:

 - Do not allow any kind of anonymous access to the system

 - We have two different profiles (internals and externals) that we differentiate with their emails domains.

 - Internal users are "normal CKAN users" that creates/updates/views datasets/resource

 - External users have limited visibility on the system and have access to only what they are granted (via collaborators or organization member)

 - Internal and External users have different home pages.

 - Accounts are managed with Azure Active Directory and SSO using ckanext-saml2auth

We only had to do a small hack overriding the User model to allow adding an `internal/external` attribute but the rest have been handled with CKAN permissions system without issues.

Bests,

**Reply by hakimsjo** [reply]
Thank you for the quick and detailed reply. It was nice to hear that you are using it internaly to.
